### ISSUE TITLE
LinkPager loose main css class before is defined

### ISSUE THREAD

**Issue by Eseperio** [yiisoft/yii2#13148]
### What steps will reproduce the problem?
Create a LinkPager widget and set only the "options" key with an empty array as value. 
`echo LinkPager::widget([
       'pagination' => $pages,
      'options'=>[
      ]
   ]);`
### What is the expected result?
Should keep "pagination" css class.
### What do you get instead?
It looses its css class.

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10
| PHP version    | 5.5.3 
| Operating system |OSX

**Comment by johonunu** [comment]
It should not keep the class, setting options to empty array basicaly removes all options.

**Comment by Eseperio** [comment]
Even if i set another parameter (like ID) it looses the class. It would be a better behavior override existing properties on demand instead removing all when options array is defined.

**Comment by dynasource** [comment]
I agree this to be cumbersome. It's better to set "default array values" in the init() method.

**Comment by SilverFire** [comment]
We do not have a declared agreement on how do we behave with properties' default value, when it is an array.

@yiisoft/core-developers we have to decide: merge or replace.

IMHO, it's better to use "replace" strategy for class configurations and "merge" for classes and options.

**Comment by Eseperio** [comment]
I think that merge should be used for the whole options array, and for each parameter should be replace, because sometimes you want to discard the default css. Keep in mind when using other visual framework rather than bootstrap you'll need custom css names.

**Comment by samdark** [comment]
I think it should replace because of reason highlighted by @Eseperio. Sometimes we need to replace classes.

**Comment by samdark** [comment]
@yiisoft/core-developers need more opinions. @klimov-paul, @cebe?

**Comment by klimov-paul** [comment]
I do not think anything should be changed. `LinkPager::$options` is a public field with default value containing CSS class:
https://github.com/yiisoft/yii2/blob/master/framework/widgets/LinkPager.php#L43

It is rather expected that passing any value for it, including empty array, will override its value entirely.

The most common use case for the `LinkPager::$options` is the `class` setup anyway. It is rare use case, which requires set other options preserving CSS class.

Since this widget does not related to any CSS framework or build-in JS code - it is not crucial for it to have container CSS class defined by default in any scenario.

Also keep in mind that any changes around `LinkPager::$options` default value is a BC break.

**Comment by Eseperio** [comment]
Why don´t use the same behavior as in [Navbar](https://github.com/yiisoft/yii2-bootstrap/blob/034d57071f7bb91074c0c2728c60ae8abdaa0e47/NavBar.php#L98) ?

**Comment by rob006** [comment]
@Eseperio Navbar is directly connected with Twitter Bootstrap - without this class it would not work as expected.

**Comment by Eseperio** [comment]
@rob006 the purpose of pagination css class is to get the bootstrap style. The main problem here is that if you want to set an id to pagination widget, or add a schema `itemprop` you are forced to declare also the class name. 
There are also a [lot of classes](https://github.com/yiisoft/yii2/blob/0b2e79eac590cf01f0985fb696da8f3c54271361/framework/widgets/LinkPager.php#L62) defined in other properties in a fixed way that are inherited from bootstrap. Because a real name for previous should be `previous` and not `prev`.

**Comment by rob006** [comment]
@Eseperio Technically framework core is not related to any CSS framework. It uses defaults compatible with Bootstrap for practical reasons, but it does not mean that it is "BootstrapLinkPager". If you change default values, it is your responsibility to make it compatible with CSS framework which you use.

**Comment by samdark** [comment]
Closing for the reasons specified.

**Comment by Eseperio** [comment]
Ok, i understand. But i was not  talking about making it compatible with other css frameworks. I was talking about why it is ok in `InputWidget` and `Pjax widget` to check if ID is set before setting default value and in the other widgets each time you customize some option need to write the css class again. When a value is a **default** it must be kept until the user overrides it, and options is not a value, is a **set of different values** wrapped into an array.  That is why in my in my humble opinion i think this behavior is wrong.

Widgets that set the whole options collection as a default:
Breadcrumb
Activefield
Detailview
LinkSorter
Listview
MaskedInput

P.D: I know ID in both widgets is used for javascript purposes.

**Comment by samdark** [comment]
OK. That needs to be unified. Will get back to it in 2.0.14.
